File: C++/ds/Fenwick_tree.cxx
#define MAX_N 100005

int fw[MAX_N];

void add(int idx,int val)
{
    while(idx<MAX_N)
    {
        fw[idx]+=val;
        idx+=(idx&-idx);
    }
}

int sum(int idx)
{
    int ret=0;
    while(idx>0)
    {
        ret+=fw[idx];
        idx-=(idx&-idx);
    }
    return ret;
}

File: Python/ds/Fenwick_tree.py
class FenwickTree:
    def __init__(self, size):
        self.size = size
        self.data = [0 for _ in range(size)]

    def add(self, idx, val):
        while idx < self.size:
            self.data[idx] += val
            idx += idx & -idx

    def sum(self, idx):
        res = 0
        while idx > 0:
            res += self.data[idx]
            idx -= idx & -idx
        return res
